private/pkg/storage/storageos/bucket.go

```diff
@@ -256,12 +256,43 @@ func (b *bucket) validateExternalPath(path string, externalPath string) error {
 		if os.IsNotExist(err) {
 			return storage.NewErrNotExist(path)
 		}
+		// The path might have a regular file in one of its
+		// elements (e.g. 'foo/bar/baz.proto' where 'bar' is a
+		// regular file).
+		//
+		// In this case, the standard library will return an
+		// os.PathError, but there isn't an exported error value
+		// to check against (i.e. os.Is*). But we can still discover
+		// whether or not this is the case by checking if any of the
+		// path components represents a regular file (e.g. 'foo/bar').
+		//
+		// It's important that we detect these cases so that
+		// multi buckets don't unnecessarily fail when one of
+		// its delegates actually defines the path.
+		elements := strings.Split(normalpath.Normalize(externalPath), "/")
+		if len(elements) == 1 {
+			// The path is a single element, so there aren't
+			// any other files to check.
+			return err
+		}
+		for i := len(elements) - 1; i >= 0; i-- {
```

Comment:
There's no way we can do this I think - this is introducing an O(file depth) operation to basically every `storageos` Bucket operation.

Reply:
Talked offline - this is just in the error case where we have a non-is-not-exist error, so not so bad